packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -411,32 +447,14 @@ export class CrossChainMessenger {
       gasLimit = migratedWithdrawalGasLimit(encoded, chainID)
       messageNonce = resolved.messageNonce
     } else {
-      const receipt = await this.l2Provider.getTransactionReceipt(
-        resolved.transactionHash
-      )
-
-      const withdrawals: any[] = []
-      for (const log of receipt.logs) {
-        if (log.address === this.contracts.l2.BedrockMessagePasser.address) {
-          const decoded =
-            this.contracts.l2.L2ToL1MessagePasser.interface.parseLog(log)
-          if (decoded.name === 'MessagePassed') {
-            withdrawals.push(decoded.args)
-          }
-        }
-      }
+      const withdrawals = await this.getWithdrawalsFromMessage(message)
 
       // Should not happen.
       if (withdrawals.length === 0) {
         throw new Error(`no withdrawals found in receipt`)
       }
 
-      // TODO: Add support for multiple withdrawals.
-      if (withdrawals.length > 1) {
-        throw new Error(`multiple withdrawals found in receipt`)
-      }
-
-      const withdrawal = withdrawals[0]
+      const withdrawal = withdrawals[multiWithdrawalIndex]
```

Comment:
This needs an undefined check

Reply:
Ah good call. I'm going to turn on a tsconfig option that makes typescript throw an error when this happens in a follow up pr